cosmos/scheduler/cmd/scheduler.go

```diff
@@ -32,39 +39,58 @@ type Scheduler struct {
 	IngestionSchedule string
 }
 
+// BucketTransaction is used as the decoded element returned by MongoDB
+type BucketTransaction struct {
+	LocationType  string
+	OperationType string `bson:"operationType"`
+	FullDocument  struct {
+		Value struct {
+			Name               string `bson:"name"`
+			OwnerDisplayName   string `bson:"ownerDisplayName"`
+			Deleted            bool   `bson:"deleted"`
+			LocationConstraint string `bson:"locationConstraint"`
+		} `bson:"value"`
+	} `bson:"fullDocument"`
+}
+
 // Run starts the Cosmos Scheduler
 func (s *Scheduler) Run() {
 	log.Println("starting cosmos scheduler")
-	quit := make(chan bool)
-	overlayUpdates, err := s.watchOverlayUpdates()
-	if err != nil {
-		log.Println(err)
-		return
-	}
 	go s.healthCheckServer()
-	err = s.configureIngestionSecret(true)
+
+	// safely create or update secret on startup
+	err := s.configureIngestionSecret(true)
 	if err != nil{
-		log.Fatal("error configuring secret:", err)
+		log.Fatal("run failed configuring secret: ", err)
 	}
+
+	overlayUpdates := s.watchOverlayUpdates(context.Background())
+	bucketCh := s.watchBucketUpdates(context.Background())
 	for {
-		go s.watchBucketUpdates(quit)
-		<-overlayUpdates
-		log.Println("received an overlay update")
-		quit <- true
+		select {
```

Comment:
@RachedBenMustapha I replaced the "watch, receive, quit" loop here with a select loop where the goroutines are no longer interdependent on each other or kill/recreated on every update. 

It's added as a separate commit to make reviewing a bit easier:
https://github.com/scality/Zenko/pull/792/commits/554002e92739824489f8725fee78a4e0df8fd926